File: DoItTest/ClientApp/src/domain/students/studentBlank.ts
import { Student } from "./student"

export class StudentBlank {
    public constructor(
        public readonly id: string | null,
        public readonly firstName: string | null,
        public readonly lastName: string | null,
        public readonly patronymic: string | null,
        public readonly group: string | null
    ) { }

    public static getDefault(): StudentBlank {
        return new StudentBlank(null, null, null, null, null)
    }

    public static formStudent(student: Student): StudentBlank {
        return new StudentBlank(student.id, student.firstName, student.lastName, student.patronymic, student.group)
    }
}

File: DoItTest.Domain/Students/StudentBlank.cs
﻿namespace DoItTest.Domain.Students
{
    public class StudentBlank
    {
        public Guid? Id { get; set; }
        public String? FirstName { get; set; }
        public String? LastName { get; set; }
        public String? Patronymic { get; set; }
        public String? Group { get; set; }

        public String FullName { get
            {
                String?[] fullName = new String?[] { LastName, FirstName, Patronymic };
                return String.Join(" ", fullName);
            } 
        }
    }
}
